[iOS] add unit tests to iOS CI jobs

xta0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30133 [iOS] add unit tests to iOS CI jobs**

### Summary

Recently we've found that the master branch was constantly broken due to some unwanted change being landed on mobile. The problem is that our CI was not able to detect the runtime errors. 

### Previous work

- Add an unit test target to the iOS TestApp ( #29962 )
- Update Fastlane to run tests ( #29963 )

### What's been changed in CI

1. XCode version has been updated to 11.2.1 
2. For iOS simulator build, we'll run some unit tests( currently only one) after the build test.

Differential Revision: [D18641413](https://our.internmc.facebook.com/intern/diff/D18641413)

xta0 committed:
[iOS] add unit test to CI

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI workflow"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI workflow"

xta0 committed:
Update on "[DO NOT MERGE][iOS] add unit test to CI workflow"

xta0 commented on .circleci/verbatim-sources/job-specs-custom.yml:
```diff
@@ -451,3 +450,26 @@
+            PROJ_ROOT=/Users/distiller/project
+            source ~/anaconda/bin/activate
+            #install the latest version of PyTorch and TorchVision
+            #TODO: How to install the nightly build?
```
@ezyang is it possible to install a PyTorch nightly build on MacOS via `pip`? I searched a lil bit, didn't find the right command to do it. The goal here is to use a recent (up-to-date) version of PyTorch to generate a torchscript model for the iOS TestApp to test.

ezyang replied:
Something like:

```
pip install --pre torch torchvision -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
```

as suggested in https://pytorch.org/get-started/locally/

ljk53 commented on .circleci/config.yml:
```diff
@@ -1302,14 +1302,14 @@ jobs:
       - should_run_job
       - checkout
-      - run_brew_for_ios_build
+      - run_brew_for_ios_build            
```
nit: remove white space?

ezyang commented on ios/TestApp/TestAppTests/TestAppTests.mm:
```diff
@@ -32,6 +31,7 @@ - (void)testForward {
   _module.eval();
   std::vector<c10::IValue> inputs;
   inputs.push_back(torch::ones({1, 3, 224, 224}, at::ScalarType::Float));
+  torch::autograd::AutoGradMode guard(false);
```
Yes, the old RAII use didn't do anything :P

ezyang commented:
I'm a little concerned about relying on nightlies for per PR tests, because it means that a botched nightly build will break all signal on this job, and you won't have any easy remediation. In this sense, it is almost better to use the official build, which will rev much less frequently. However, the official build is only really OK if we maintain backwards-compatibility (which we should be, right @houseroad ?)  So I feel the patch as is is better.

xta0 committed:
Update on "[iOS] add unit tests to iOS CI jobs"


### Summary

Recently we've found that the master branch was constantly broken due to some unwanted change being landed on mobile. The problem is that our CI was not able to detect the runtime errors. 

### Previous work

- Add an unit test target to the iOS TestApp ( #29962 )
- Update Fastlane to run tests ( #29963 )

### What's been changed in CI

1. XCode version has been updated to 11.2.1 
2. For iOS simulator build, we'll run some unit tests( currently only one) after the build test.

Differential Revision: [D18641413](https://our.internmc.facebook.com/intern/diff/D18641413)

xta0 committed:
Update on "[iOS] add unit tests to iOS CI jobs"


### Summary

Recently we've found that the master branch was constantly broken due to some unwanted change being landed on mobile. The problem is that our CI was not able to detect the runtime errors. 

### Previous work

- Add an unit test target to the iOS TestApp ( #29962 )
- Update Fastlane to run tests ( #29963 )

### What's been changed in CI

1. XCode version has been updated to 11.2.1 
2. For iOS simulator build, we'll run some unit tests( currently only one) after the build test.

Differential Revision: [D18641413](https://our.internmc.facebook.com/intern/diff/D18641413)

xta0 committed:
Update on "[iOS] add unit tests to iOS CI jobs"


### Summary

Recently we've found that the master branch was constantly broken due to some unwanted change being landed on mobile. The problem is that our CI was not able to detect the runtime errors. 

### Previous work

- Add an unit test target to the iOS TestApp ( #29962 )
- Update Fastlane to run tests ( #29963 )

### What's been changed in CI

1. XCode version has been updated to 11.2.1 
2. For iOS simulator build, we'll run some unit tests( currently only one) after the build test.

Differential Revision: [D18641413](https://our.internmc.facebook.com/intern/diff/D18641413)

xta0 committed:
Update on "[iOS] add unit tests to iOS CI jobs"


### Summary

Recently we've found that the master branch was constantly broken due to some unwanted change being landed on mobile. The problem is that our CI was not able to detect the runtime errors. 

### Previous work

- Add an unit test target to the iOS TestApp ( #29962 )
- Update Fastlane to run tests ( #29963 )

### What's been changed in CI

1. XCode version has been updated to 11.2.1 
2. For iOS simulator build, we'll run some unit tests( currently only one) after the build test.

Differential Revision: [D18641413](https://our.internmc.facebook.com/intern/diff/D18641413)

hlu1 commented:
@xta0, I'm reverting this PR because it broke master. I saw that the fix #30327 is on the way. Please make sure it doesn't break other jobs.
